dev/ci/internal/ci/executor_operations.go

```diff
@@ -146,7 +146,7 @@ func executorImageFamilyForConfig(c Config) string {
 		if err != nil {
 			panic("cannot parse version")
 		}
-		imageFamily = fmt.Sprintf("sourcegraph-executors-internal-%d-%d-%d", ver.Major(), ver.Minor(), ver.Patch())
+		imageFamily = fmt.Sprintf("sourcegraph-executors-internal-%d-%d", ver.Major(), ver.Minor())
```

Comment:
do we ever actually want these internal names? 

```suggestion
		imageFamily = fmt.Sprintf("sourcegraph-executors-%d-%d", ver.Major(), ver.Minor())
```

And similar for the docker-mirror. If I understood correctly.

Reply:
I did read the PR description btw, I'm just not getting why 😅

Reply:
Yes we do - **without** internal in the image family name an executor meant for internal releases *only* will be published with the image family name `sourcegraph-executors` which means the terraform in `terraform-aws` and `terraform-google` who look for latest images in the image family `sourcegraph-executors-<MAJOR>-<MINOR>` will pick the image up _after_ every `MAJOR.MINOR` bump.

So ultimately, internal releases are kept to the `sourcegraph-executors-internal` and once you promote to public the image is rebuilt and put into `sourcegraph-executors` which means the released terraform will now pick it up.

This is also has a nice quality which I think the release team can investigate. This technically allows us to test internal releases _with the terraform. If you want to opt into an internal release you just have to update the image family in the terraform - almost like `alpha`, `beta` changes.

Sorry if I am over explaining it! Not my intention at all.

The major caveat not mentioned here are at all, is that you kinda just have to _know_ is that you'll only see the `sourcegraph-executor-MAJOR-MINOR` release once *the tagged release build finishes* which flies completely under the radar - which I think we need to fix somehow.

One idea I just got on how this _can_ be done is part of the promote to public release steps in the release manifest is to run:
```bash
# promote to public 
gcloud compute images create --source-image=sourcegraph-executors-internal-MAJOR-MINOR-BUILD sourcegraph-executors-MAJOR-MINOR-4-BUILD --family sourcegraph-executors-MAJOR-MINOR
```

This stays _more_ true to the spirit of promotion, in that a new image is not built by CI, instead the image used for internal releases just gets copied to a new name.

Reply:
Awesome 👏🏼 

Yup - this should be backported ♻️